src/__tests__/Banner.js

```diff
@@ -0,0 +1,191 @@
+    )
+    expect(json).toHaveStyleRule('color', theme.palette.blue.color)
+  })
+
+  test('renders with green palette', () => {
+    const json = renderer
+      .create(<Banner palette="green" theme={theme} />)
+      .toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule(
+      'background-color',
+      theme.palette.green.backgroundColor
+    )
+    expect(json).toHaveStyleRule('color', theme.palette.green.color)
+  })
+
+  test('renders with orange palette', () => {
+    const json = renderer
+      .create(<Banner palette="orange" theme={theme} />)
+      .toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule(
+      'background-color',
+      theme.palette.orange.backgroundColor
+    )
+    expect(json).toHaveStyleRule('color', theme.palette.orange.color)
+  })
+
+  test('renders with red palette', () => {
+    const json = renderer
+      .create(<Banner palette="red" theme={theme} />)
+      .toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule(
+      'background-color',
+      theme.palette.red.backgroundColor
+    )
+    expect(json).toHaveStyleRule('color', theme.palette.red.color)
+  })
+
+  test('renders with lightBlue palette', () => {
+    const json = renderer
+      .create(<Banner palette="lightBlue" theme={theme} />)
+      .toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule(
+      'background-color',
+      theme.palette.lightBlue.backgroundColor
+    )
+    expect(json).toHaveStyleRule('color', theme.palette.lightBlue.color)
+  })
+
+  test('renders with lightGreen palette', () => {
+    const json = renderer
+      .create(<Banner palette="lightGreen" theme={theme} />)
+      .toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule(
+      'background-color',
+      theme.palette.lightGreen.backgroundColor
+    )
+    expect(json).toHaveStyleRule('color', theme.palette.lightGreen.color)
+  })
+
+  test('renders with lightOrange palette', () => {
+    const json = renderer
+      .create(<Banner palette="lightOrange" theme={theme} />)
+      .toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule(
+      'background-color',
+      theme.palette.lightOrange.backgroundColor
+    )
+    expect(json).toHaveStyleRule('color', theme.palette.lightOrange.color)
+  })
+
+  test('renders with lightRed palette', () => {
+    const json = renderer
+      .create(<Banner palette="lightRed" theme={theme} />)
+      .toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule(
+      'background-color',
+      theme.palette.lightRed.backgroundColor
+    )
+    expect(json).toHaveStyleRule('color', theme.palette.lightRed.color)
+  })
+
+  test('renders close button if onClose func is provided', () => {
+    const wrapper = shallow(<Banner onClose={() => {}} theme={theme} />)
+    const close = wrapper.find('CloseButton')
+    expect(close).toHaveLength(1)
```

Comment:
I think this sort of thing is okay here, but when I see a lot of conditional logic checking for children, I think it's good to consider, *should this be another component* or *can this simply be handled with composition*

An alternative API for a component like this could look like:

```jsx
<Banner color='blue'>
  <Icon name='check' />
  <Text>Hello</Text>
  <CloseButton onClick={this.handleClose} />
</Banner>
```

Using `React.Children` you can generally handle any internal styling that you might need for something like this